builder - templates - overview (#196)
moondance-labs/tanssi-docs: builders/build/templates/overview.md

---
title: Templates

> description: Tanssi includes templates to kick-start the development of an appchain offering options such as a basic one and another featuring full EVM (Ethereum) support.

Reviewer: ```suggestion
description: Tanssi includes templates to kick-start the development of an appchain with full EVM (Ethereum) support and minimal engineering overhead.
```

Reviewer: I think it was discussed that we just mention the EVM template btw (considering the previous PRs we merged)

---

# Templates Overview {: #templates-overview }

## Introduction {: #introduction }

> Appchains deployed through Tanssi are fully customizable blockchains benefiting from a shared set of sequencers and the security of a provider of their choice ([Symbiotic](https://symbiotic.fi/){target=\_blank}, for example). They also need to implement the necessary functionalities and configurations to support the Tanssi protocol, which the templates presented in this article already do, making it easier for developers.

Reviewer: ```suggestion
Appchains deployed through Tanssi are fully customizable blockchains benefiting from a shared set of sequencers and the security of a provider of their choice (for example, [Symbiotic](https://symbiotic.fi/){target=\_blank} for Ethereum). They also need to implement the necessary functionalities and configurations to support the Tanssi protocol, which the templates presented in this article already do, making it easier for developers.
```

As presented in the [Included Templates](/learn/tanssi/included-templates/#baseline-appchain-template){target=\_blank} from the Learn section, Tanssi already provides two templates to jumpstart the development process:

> - **[Baseline Tanssi appchain template](/learn/tanssi/included-templates/#baseline-appchain-template){target=\_blank}** - a template that provides the basic platform to start adding custom logic 

Reviewer: Maybe we remove? Unsure

Besides block production, there are other essential aspects for any appchain covered in the templates, such as:

- **Consensus** - appchains have the necessary functionality to allow the sequencers to produce blocks, gossip and validate them, and coordinate with the security provider to get notified about the block's finality
- **Appchains Interoperability** - handles the ingestion and dispatch of incoming downward and lateral messages, allowing a Tanssi appchain to communicate and interoperate with the other chains within the ecosystem
- **Runtime Upgrades** - a runtime upgrade in a Tanssi appchain must be informed to the security provider's operators to allow them to check on the blocks produced by the sequencers of the Tanssi appchains

## Included Modules {: #included-modules }

Besides the necessary modules to support the operation of a Tanssi appchain, many other modules provide functional behavior that the users can interact with. 

These are some of the functional modules exposing a behavior to the users that are included in the templates and ready to use:

> - **[pallet_balances](https://paritytech.github.io/substrate/master/pallet_balances/index.html){target=\_blank}** - the Balances pallet provides functions for handling accounts and balances for the Tanssi appchain native currency

Reviewer: Call them pallets still?

Author: Nope, I missed those!
